apps/src/netsim/netsim.js

```diff
@@ -304,7 +322,9 @@ NetSim.prototype.changeRemoteDnsMode = function (newDnsMode) {
  * @param {boolean} isDnsNode
  */
 NetSim.prototype.setIsDnsNode = function (isDnsNode) {
-  this.tabs_.setIsDnsNode(isDnsNode);
```

Comment:
I've been trying to work toward making UI components (like the tabs and their subcomponents) blind to the outside world, in case I'm eventually asked to move stuff over to React.js - so instead of letting them do global listening, the top-level `NetSim` is the dispatcher for changes, propagating them down the chain of UI components in this way.  It does mean having some unfortunate boilerplate, but it also means no tracking of who is listening to whom.

That said, `isDnsNode` is an exemplar of state that is, for practical purposes, global.  There are many such settings.  Would it be cleaner to actually treat it as a global, living on a singleton that low-level components could watch (assured that it won't go away) and respond to changes on?